pkg/api/types.go

```diff
@@ -1407,6 +1409,69 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// RateLimit is a rate limit on disruptions (preemption and other evictions) across the corresponding
+	// collection of pods, e.g. no more than one disruption per hour across the pods of a particular Service
+	RateLimit *RateLimitSpec `json:"ratelimit,omitempty"`
+	// The maximum number of pods that can be unavailable simultaneously.
+	// Value can be an absolute number (ex: 5) or a percentage of total pods(ex: 10%).
+	// Absolute number is calculated from percentage by rounding up.
+	// By default, a fixed value of 1 is used.
+	MaxUnavailable intstr.IntOrString `json:"maxUnavailable,omitempty"`
+	// Minimum number of seconds for which a newly created pod should be ready
+	// without any of its container crashing, for it to be considered available.
+	// Defaults to 0 (pod will be considered available as soon as it is ready)
+	MinReadySeconds int `json:"minReadySeconds,omitempty"`
```

Comment:
I understand why this is here, but I worry that the conjunction of this, activeDeadlineSeconds in the pod spec, the other assorted timeouts in the pod probe spec, and MinReadySeconds in deployment is confusing.

It would be nice if we could reduce the number of places the user had to configure these related timeouts.  Any ideas?